### ISSUE TITLE
define type="numeric" by giving renderer, editor and validator

### ISSUE THREAD

**Issue by stla** [handsontable/handsontable#4054]
Hello,

 I [figured out](http://stackoverflow.com/questions/42025001/strange-behaviour-of-comma-as-decimal-separator-in-handsontable) that `numbro(x).format()` unformats `x` when `x` is not valid. This sounds unexpected and nonsensical to me (getting an unformatting when one requires a formatting...)

Therefore, I tried to write a custom `type="numeric"` with another formatting.

And I realized that, contrary to the claim written [in the doc here](http://docs.handsontable.com/0.30.0/tutorial-cell-function.html), it is *not* true that `type="numeric"` is equivalent to 
```
{
      renderer: Handsontable.NumericRenderer,
      editor: Handsontable.editors.TextEditor,
      validator: Handsontable.NumericValidator
}
```

Indeed, try this fiddle: https://jsfiddle.net/stla/wj5yy0zu/6/

If you modifiy (edit) cell (0,0), say you type the number `14`, then the new value is the number `14` as expected.

But if you modifiy cell (1,1) and type the number `14`, then the new value is the **string** `"14"`: 

![capture](https://cloud.githubusercontent.com/assets/4466543/22639577/f110f8fa-ec4e-11e6-8981-ed83836bada0.PNG)

I would appreciate a hint to get numbers after editing.

**Comment by AMBudnik** [comment]
@stla thanks for sharing, you are right there is a `NumericEditor`

I've updated your demo here: https://jsfiddle.net/handsoncode/Ldm14epp/

I set the `NumericEditor` for the second column but get the typeof result `string`. We should definitely update this tutorial.

**Comment by stla** [comment]
Thank you @AMBudnik 
Yes, I already tried `NumericEditor` but this doesn't change anything.
So you mean that currently you have no solution to get a number ?

**Comment by stla** [comment]
And if I remember well, I also tried that in the `columns` setting:
```
type: "numeric",
renderer: myNumericRenderer
validator: myNumericValidator
```
(since I only want to modifiy the renderer and the validator). But (again, if I remember well), this still converts the numbers into strings. I'll try again since I'm not sure to rememember well.

**Comment by stla** [comment]
Also, I don't know whether you agree, but `numbro(x).format()` is nonsensical (I mean the behavior I show in my SO post). I will open an issue on the numbro github repo about that.

**Comment by stla** [comment]
Even when I set type `"mynumeric"` and I do:
```
mynumericCell = Handsontable.NumericCell; 
Handsontable.cellTypes = {
  text: Handsontable.TextCell,
  date: Handsontable.DateCell,
  numeric: Handsontable.NumericCell,
  mynumeric: mynumericCell,
  checkbox: Handsontable.CheckboxCell,
  autocomplete: Handsontable.AutocompleteCell,
  handsontable: Handsontable.HandsontableCell
};
```
then after editing, numbers are transformed to strings.

**Comment by stla** [comment]
Yes, my above remark is normal, because of this line in the function `validateChanges`:
```
        if (cellProperties.type === 'numeric' && typeof changes[i][3] === 'string') {
```

**Comment by stla** [comment]
... and this also explains the problem raised in my OP.

**Comment by AMBudnik** [comment]
Hi @stla 

I have discussed this issue with our developer Piotr. The `NumericEditor` has no additional logic and it calls `TextEditor` for handling input so the docs are right.

When it comes to the string you're getting it is a string cause we miss the meta data settings that define a cell as a numeric. So basically all this settings are bonded together and you need to add type to your cells to make then `umeric`.

**Comment by stla** [comment]
Ok, thank you. This happens because of the test `cellProperties.type === 'numeric'` in the function `validateChanges`.
I don't like what happens in `validateChanges` when the condition `cellProperties.type === 'numeric' && typeof changes[i][3] === 'string'` is `true` (there's an unfortunate unformatting). So I had to modify the source code at this place to get what I want, since there's no other way to control that, I think.

Thank you for your help.

**Comment by stla** [comment]
> so the docs are right

Not really, there is no equivalence, for the above mentioned reason. If you don't put `type='numeric'` then the condition `cellProperties.type === 'numeric' && typeof changes[i][3] === 'string'` is never `true`.

**Comment by AMBudnik** [comment]
Yes, I guess that we can change it a bit, as in the end of the page at `Cell functions getters` it is written that `type` is a part of needed `cellProperties` and by reading only above information it looks like setting: editor, validator and renderer does the same job.
I'll mark this issue as a 'documentation needed' to rewrite it in future.

Thanks for sharing and for your investigation @stla
